scripts/azureml-assets/azureml/assets/validate_tree.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Validate source tree files."""
+
+import argparse
+import sys
+from pathlib import Path
+from typing import List
+
+import azureml.assets.util as util
+from azureml.assets.util import logger
+
+BYTES_IN_A_MEGABYTE = 1024 ** 2
+
+
+def validate_tree(input_dirs: List[Path]):
+    """Validate source tree files.
+
+    Args:
+        input_dirs (List[Path]): List of directories to search for assets.
+
+    Returns:
+        bool: True if files were successfully validated, otherwise False.
+    """
+    error_count = 0
+
+    # Check that every spec.yaml has asset.yaml next to it
+    for spec_config_file in util.find_files(input_dirs, "spec.yaml"):
+        asset_config_file = spec_config_file.parent / "asset.yaml"
+        if not asset_config_file.exists():
+            logger.log_error(f"{spec_config_file} does not have a corresponding asset.yaml")
+            error_count += 1
+
+    # Fail if any asset.yml in the tree
+    for asset_yml_file in util.find_files(input_dirs, "asset.yml"):
+        logger.log_error(f"{asset_yml_file} should be named asset.yaml")
+        error_count += 1
+
+    # Scan every file in the source tree to be 1 MB or less
+    for file in util.find_files(input_dirs, "*"):
```

Comment:
What would you think about moving all of the other checks inside this loop so the directory tree only needs to be traversed once? [`Path.name`](https://docs.python.org/3/library/pathlib.html) gives you the name of the file, excluding its path.

Reply:
I like this approach as it'll still allow for each check to be done separately in this for-loop